contracts/VeJoeStaking.sol

```diff
@@ -0,0 +1,267 @@
+
+    /// @notice Initialize with needed parameters
+    /// @param _joe Address of the JOE token contract
+    /// @param _veJoe Address of the veJOE token contract
+    /// @param _baseGenerationRate Rate of veJOE generated per sec per JOE staked
+    /// @param _boostedGenerationRate Boosted rate of veJOE generated per sec per JOE staked
+    /// @param _boostedThreshold Percentage of total staked JOE user has to deposit to be boosted
+    /// @param _boostedDuration Length of time a user receives boosted benefits
+    function initialize(
+        IERC20Upgradeable _joe,
+        VeJoeToken _veJoe,
+        uint256 _baseGenerationRate,
+        uint256 _boostedGenerationRate,
+        uint256 _boostedThreshold,
+        uint256 _boostedDuration
+    ) public initializer {
+        require(address(_joe) != address(0), "VeJoeStaking: unexpected zero address for _joe");
+        require(address(_veJoe) != address(0), "VeJoeStaking: unexpected zero address for _veJoe");
+        require(
+            _boostedGenerationRate > _baseGenerationRate,
+            "VeJoeStaking: expected _boostedGenerationRate to be greater than _baseGenerationRate"
+        );
+        require(_boostedThreshold <= 100, "VeJoeStaking: expected _boostedThreshold to be less than or equal to 100");
+
+        __Ownable_init();
+
+        maxCap = 100;
+        joe = _joe;
+        veJoe = _veJoe;
+        baseGenerationRate = _baseGenerationRate;
+        boostedGenerationRate = _boostedGenerationRate;
+        boostedThreshold = _boostedThreshold;
+        boostedDuration = _boostedDuration;
+    }
+
+    /// @notice Set maxCap
+    /// @param _maxCap the new maxCap
+    function setMaxCap(uint256 _maxCap) external onlyOwner {
+        require(_maxCap > 0, "VeJoeStaking: expected new _maxCap to be greater than 0");
+        maxCap = _maxCap;
+    }
+
+    /// @notice Set baseGenerationRate
+    /// @param _baseGenerationRate the new baseGenerationRate
+    function setBaseGenerationRate(uint256 _baseGenerationRate) external onlyOwner {
+        require(
+            _baseGenerationRate < boostedGenerationRate,
+            "VeJoeStaking: expected new _baseGenerationRate to be less than boostedGenerationRate"
+        );
+        baseGenerationRate = _baseGenerationRate;
+    }
+
+    /// @notice Set boostedGenerationRate
+    /// @param _boostedGenerationRate the new boostedGenerationRate
+    function setBoostedGenerationRate(uint256 _boostedGenerationRate) external onlyOwner {
+        require(
+            _boostedGenerationRate > baseGenerationRate,
+            "VeJoeStaking: expected new _boostedGenerationRate to be greater than baseGenerationRate"
+        );
+        boostedGenerationRate = _boostedGenerationRate;
+    }
+
+    /// @notice Set boostedThreshold
+    /// @param _boostedThreshold the new boostedThreshold
+    function setBoostedThreshold(uint256 _boostedThreshold) external onlyOwner {
+        require(
```

Comment:
as per above comment we also need a > 0 check.

Reply:
See my comment (https://github.com/traderjoe-xyz/joe-core/pull/29#discussion_r799953113)